packages/gatsby/src/utils/start-server.ts

```diff
@@ -190,11 +190,30 @@ export async function startServer(
    * If no GATSBY_REFRESH_TOKEN env var is available, then no Authorization header is required
    **/
   const REFRESH_ENDPOINT = `/__refresh`
-  const refresh = async (req: express.Request): Promise<void> => {
+  const refresh = async (
+    req: express.Request,
+    pluginName?: string
+  ): Promise<void> => {
     emitter.emit(`WEBHOOK_RECEIVED`, {
       webhookBody: req.body,
+      pluginName,
     })
   }
+
+  app.post(`${REFRESH_ENDPOINT}/:plugin_name`, (req, res) => {
```

Comment:
Feels like this should only be conditionally, when such a plugin is actually used

Reply:
Perhaps, although exposing the endpoint unconditionally seems harmless?

Reply:
Wouldn't it be possible to hammer Cloud preview this way?

Reply:
AIUI preview pods should skip intermediary builds. We can throttle the actual webhook. Does it seem worse somehow than the regular /__refresh or are you concerned about that one as well?

Reply:
Also I'm out until 11/30

Reply:
Yeah those would worry me as well. I'll raise it.

Reply:
Aren't build webhooks on Cloud sort of secret? They're separate from the Preview frontend url.

Reply:
Yeah, the CMS will have a webhook URL and a preview token. The webhook URL goes to captain-hook and we can throttle that.

Reply:
> Aren't build webhooks on Cloud sort of secret? They're separate from the Preview frontend url.

Not anymore ;)